juju/terraform-provider-juju
# [JUJU-3791] GitHub actions running Microk8s

## Description

We need to setup an environment where we can test operations in a K8s environment. This PR extends the idea of the matrix to include the `TEST_CLOUD` environment variable to deploy a Microk8s controller and run the corresponding tests if any.

Additionally, I have reduced the number of terraform versions to be tested to 1.3, 1.4, and 1.5.

## Type of change

Please mark if proceeds.


## Environment
NA

## QA steps

Wait for Github actions to run successfully. Observe that the `TestAcc_ResourceApplication_Basic` skips.

# Additional notes

There is an ongoing issue with the `TestAcc_ResourceAccessModel_Basic` when using microk8s in GitHub actions there is a problem with the k8s connection. Interestingly, I cannot reproduce this locally. I have skipped this test for the moment if no LXD cloud is set. However, this has to be investigated.

Additional testing for K8s deployments will be addressed in other PRs.
